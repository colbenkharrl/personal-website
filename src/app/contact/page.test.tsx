import { render, screen } from '@testing-library/react';
import Page from './page';

jest.mock('next/navigation', () => ({
  ...jest.requireActual('next/navigation'),
  useRouter: jest.fn(),
}));
jest.mock('react-dom', () => ({
  ...jest.requireActual('react-dom'),
  useFormStatus: () => ({
    pending: false,
  }),
}));

it('should match page snapshot', () => {
  render(<Page />);

  expect(screen.getByTestId('Container')).toMatchSnapshot('page snapshot');
});
