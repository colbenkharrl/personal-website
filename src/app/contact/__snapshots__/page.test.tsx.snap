// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match page snapshot: page snapshot 1`] = `
<div
  class="sm:px-8 mt-16 sm:mt-32"
  data-testid="Container"
>
  <div
    class="mx-auto w-full max-w-7xl lg:px-8"
  >
    <div
      class="relative px-4 sm:px-8 lg:px-12"
    >
      <div
        class="mx-auto max-w-2xl lg:max-w-5xl"
      >
        <header
          class="max-w-2xl"
        >
          <h1
            class="text-4xl font-bold tracking-tight text-zinc-800 dark:text-zinc-100 sm:text-4xl"
          >
            Get in touch with me
          </h1>
          <p
            class="mt-6 text-base text-zinc-600 dark:text-zinc-400"
          >
            Use the form below to get in contact with me. I try to respond as soon as I am able, usually within one week.
          </p>
        </header>
        <div
          class="mt-16 sm:mt-20"
        >
          <form
            data-testid="ContactForm"
          >
            <div
              class="space-y-12"
            >
              <div
                class="border-b border-gray-900/10 pb-12"
              >
                <div
                  class="mt-10 grid grid-cols-1 gap-x-6 gap-y-8 sm:grid-cols-6"
                >
                  <div
                    class="sm:col-span-3"
                  >
                    <label
                      class="block text-sm font-medium leading-6 text-zinc-800 dark:text-zinc-200"
                      for="firstName"
                    >
                      First name
                      <span
                        class="text-red-500"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="relative mt-2 rounded-md shadow-sm"
                    >
                      <input
                        aria-invalid="false"
                        autocomplete="givenName"
                        class="block w-full rounded-md border-0 bg-transparent py-1.5 text-zinc-800 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-yellow-500  dark:text-zinc-200 dark:focus:ring-yellow-400 sm:text-sm sm:leading-6"
                        id="firstName"
                        name="firstName"
                        placeholder="Jane"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="sm:col-span-3"
                  >
                    <label
                      class="block text-sm font-medium leading-6 text-zinc-800 dark:text-zinc-200"
                      for="lastName"
                    >
                      Last name
                      <span
                        class="text-red-500"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="relative mt-2 rounded-md shadow-sm"
                    >
                      <input
                        aria-invalid="false"
                        autocomplete="familyName"
                        class="block w-full rounded-md border-0 bg-transparent py-1.5 text-zinc-800 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-yellow-500  dark:text-zinc-200 dark:focus:ring-yellow-400 sm:text-sm sm:leading-6"
                        id="lastName"
                        name="lastName"
                        placeholder="Doe"
                        type="text"
                      />
                    </div>
                  </div>
                  <div
                    class="col-span-full"
                  >
                    <label
                      class="block text-sm font-medium leading-6 text-zinc-800 dark:text-zinc-200"
                      for="email"
                    >
                      Email address
                      <span
                        class="text-red-500"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="relative mt-2 rounded-md shadow-sm"
                    >
                      <input
                        aria-invalid="false"
                        autocomplete="email"
                        class="block w-full rounded-md border-0 bg-transparent py-1.5 text-zinc-800 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-yellow-500  dark:text-zinc-200 dark:focus:ring-yellow-400 sm:text-sm sm:leading-6"
                        id="email"
                        name="email"
                        placeholder="jdoe@email.com"
                        type="email"
                      />
                    </div>
                  </div>
                  <div
                    class="col-span-full"
                  >
                    <label
                      class="block text-sm font-medium leading-6 text-zinc-800 dark:text-zinc-200"
                      for="message"
                    >
                      Message
                      <span
                        class="text-red-500"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="relative mt-2 rounded-md shadow-sm"
                    >
                      <textarea
                        aria-invalid="false"
                        class="block w-full rounded-md border-0 bg-transparent px-2 py-1.5 text-zinc-800 shadow-sm ring-1 ring-inset ring-gray-300 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-yellow-500 dark:text-zinc-200 dark:focus:ring-yellow-400 sm:text-sm sm:leading-6"
                        id="message"
                        name="message"
                        placeholder="I would like to chat about..."
                        rows="3"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="mt-6 flex items-center justify-start gap-x-6"
                >
                  <button
                    class="disabled:opacity-30 inline-flex items-center gap-2 justify-center rounded-md py-2 px-3 text-sm outline-offset-2 transition active:transition-none bg-yellow-500 font-semibold text-zinc-100 hover:bg-yellow-600 active:bg-yellow-500 active:text-zinc-100/70 dark:bg-yellow-400 dark:hover:bg-yellow-500 dark:active:bg-yellow-700 dark:active:text-zinc-100/70 w-full"
                    type="submit"
                  >
                    Submit
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
